tsukuba2023/scripts/movingbase_node.py

```diff
@@ -0,0 +1,119 @@
+import serial
+import rospy
+from sensor_msgs.msg import Imu
+from std_msgs.msg import Float64
+import math
+import numpy as np
+
+nowPoint=[0]*10
+HEADER = 6
+count = 0
+first_heading= 0
+
+
+def readPosned():
+    ackPacket=[b'\xB5',b'\x62',b'\x01',b'\x3C',b'\x00',b'\x00']
+    i = 0
+    payloadlength = 6
+    with serial.Serial('/dev/sensors/GNSSrover', 19200, timeout=1) as ser:
+        while i < payloadlength+8: 
+            incoming_char = ser.read()         
+            if (i < 3) and (incoming_char == ackPacket[i]):
+                i += 1
+            elif i == 3:
+                ackPacket[i]=incoming_char
+                i += 1           
+            elif i == 4 :
+                ackPacket[i]=incoming_char
+                i += 1
+            elif i == 5 :
+                ackPacket[i]=incoming_char        
+                payloadlength = int.from_bytes(ackPacket[4]+ackPacket[5], byteorder='little',signed=False)
+                i += 1
+            elif (i > 5):
+                ackPacket.append(incoming_char)
+                i += 1
+    if checksum(ackPacket,payloadlength) :
+        perseheading(ackPacket)
+
+
+def checksum(ackPacket,payloadlength ):
+    CK_A =0
+    CK_B =0
+    for i in range(2, payloadlength+6):
+        CK_A = CK_A + int.from_bytes(ackPacket[i], byteorder='little',signed=False) 
+        CK_B = CK_B +CK_A
```

Comment:
フォーマットについて、例えばこんな感じです。
43行目の +6 も意味があるなら変数の方がいいです。

```suggestion
def checksum(ackPacket, payloadlength):
    CK_A = 0
    CK_B = 0
    for i in range(2, payloadlength+6):
        CK_A = CK_A + int.from_bytes(ackPacket[i], byteorder='little', signed=False) 
        CK_B = CK_B + CK_A
```